File: src/types/work-item.ts
export type WorkItemKind = "project" | "requirement" | "task" | "subtask";

export type WorkItem = {
  id: number;
  itemId: string;
  kind: WorkItemKind;
  parentId?: number;
  title: string;
  status: string;
  priority: string;
  owner: string;
  content: string;
  effort?: number;
  planMonth?: string;
  plannedHours?: number;
  actualHours?: number;
  dueDate?: string;
  updatedAt: number;
};

export type WorkItemListQuery = {
  page: number;
  pageSize: number;
  kind?: WorkItemKind;
  parentId?: number;
  keyword?: string;
  status?: string;
  priority?: string;
  sortField?: "updatedAt" | "title";
  sortOrder?: "ascend" | "descend";
};

export type WorkItemListResult = {
  items: WorkItem[];
  total: number;
  page: number;
  pageSize: number;
};

export type CreateWorkItemPayload = {
  kind: WorkItemKind;
  parentId?: number;
  title: string;
  status: string;
  priority: string;
  owner: string;
  content?: string;
  effort?: number;
  planMonth?: string;
  plannedHours?: number;
  actualHours?: number;
  dueDate?: string;
};

export type UpdateWorkItemPayload = Omit<CreateWorkItemPayload, "kind"> & {
  id: number;
};

export type WorkItemParentOption = {
  id: number;
  itemId: string;
  title: string;
};

File: src-tauri/src/models/work_item.rs
use sea_orm::entity::prelude::*;

#[derive(Clone, Debug, PartialEq, DeriveEntityModel)]
#[sea_orm(table_name = "work_item")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub item_id: String,
    pub kind: String,
    pub parent_id: Option<i32>,
    pub title: String,
    pub status: String,
    pub priority: String,
    pub owner: String,
    pub content: String,
    pub effort: Option<f64>,
    pub plan_month: Option<String>,
    pub planned_hours: Option<f64>,
    pub actual_hours: Option<f64>,
    pub due_date: Option<String>,
    pub updated_at: i64,
    pub created_at: i64,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
